### DISCUSSION TITLE
Editing discussion Dec 2023

### DISCUSSION THREAD

**Discussion by editdare** [FixOurFood/FoF_streamlit_dashboard#22] [General]
For discussions around checking text and functionality, Dec 2023.

**Comment by jucordero** [comment]
Thanks for opening this discussion, @editdare.
For future reference, here is a list of resources associated with the dashboard.

**Tooltips**: Each slider is associated a tooltip text which is read from an external spreadsheet:
https://docs.google.com/spreadsheets/d/1Mm2Cg6Ofz4W08RBjxKgf5mLBp7u3F3sFy0HgYYd8r3w/edit?usp=sharing

Changes to this spreadsheet should appear after a couple minutes in the main app. We have numbered the fields to make it easier to locate which cell corresponds to each widget in the streamlit app 

**Modelling document**: Each slider and selection box is meant to represent the 'ambition level' associated with a food system intervention. In some cases the ambition level is a percentage, in some other cases it can be an absolute value.
Our modelling document explains what each slider does and how it translates into the different charts and metrics.
It is a bit outdated, but the idea is to use it as the official knowledge resource of the Dashboard
https://docs.google.com/document/d/1A2J4BYIuXMgrj9tuLtIon8oJTuR1puK91bbUYCI8kHY/edit#heading=h.4ajpofiquxi

**AgriFoodPy**: This is the modelling package we plan to use under the hood. It contains all the functions to manipulate tabular data, calculate effects of interventions and plot data.
We are in the process of migrating the code to use AgriFoodPy functionality. This won't affect the dashboard in any way, only make the code more legible and organized.
We wrote documentation for the code, which can be found here in case needed:
https://agrifoodpy.readthedocs.io/en/latest/

**Reply by editdare** [reply]
Hi @jucordero 
I'd like to check as we start out that the system we're using to talk about different aspects of the dashboard is helpful for all. So please have a look at the following and let me know what you think:

**Dashboard edits/troubleshooting for specific items:** Here's a spreadsheet of comments that has space for responses from you or others (responses could be initialed to track who's who):
https://docs.google.com/spreadsheets/d/19KAdjIa6py2gVk5DrfFgqvmMr-jG1uvvVR0mz_LKF7c/edit?usp=sharing

**Wider queries:** Some queries might span more than one item, or be wider ranging or longer term. Here is a separate document to show some examples. As these may produce conversations, it may be best to use a new discussion thread for each - but I'd like to know what you think.
https://docs.google.com/document/d/123ddLU9GXipYWClDLc7kJN7SsCB9tnfURGSIzCz9ehs/edit?usp=sharing

Once we've got the system up and running I will continue to check the dasboard and documentation.

**Reply by jucordero** [reply]
Hi @editdare 
Thanks a lot for this! This is a great way to keep track of suggestions for the dashboard.

Regarding the troubleshooting, you have highlighted some errors I had frogot about. Thanks for this.
I suspect some of these errors (like the ones when you select certain item groups) will go away once I port the code to the latest version of AgriFoodPy, which I'm currently working on.

I will go through the wider queries on the document and once we have agreed on the changes to be made, I will add the corresponding "issues" to this GitHub repository so I can keep track of the necessary code changes.

Thank you so much!

**Reply by jucordero** [reply]
Hi @editdare 

I've added 4 new entries to the Issues tracker to start working on the changes you have suggested.

- #24 deals with the errors when selecting certain item groups in the "CO2 emissions per food item" sub menu.
- #25 for all suggestions related to the behaviour of reset buttons
- #27 for all visible text suggestions, including text and formatting
- #28 To address the issue of "Meat-free days" and "Ruminant meat" sliders not behaving as expected.

I will start making changes on the "develop" branch before merging to the main branch. To preview the changes to the main dashboard I have set up a development version where we can test things. 

https://fixourfood-develop.streamlit.app/

I have added myself to the "Who to action" column on all rows that mention these newly created issues as they depend on changes to the Python code.
Nevertheless, if you're feeling adventurous you can try editing the code yourself to fix the issues in #27 as they only require editing some strings. Just remember to edit the **develop** branch instead of **main**.

The `tooltip` type suggestions must be addressed by editing the tooltip spreadsheet directly. Feel free to make the changes yourself. It can take up to 5 minutes for changes to be reflected in the dashboard. These will affect the **main** and **develop** dashboards simultaneously. That is fine, as changes in the spreadsheet are much easier to revert.

**Reply by editdare** [reply]
Hi @jucordero 
I've just added two additional comments to earlier rows (using column J), and added one new row (bottom) to https://docs.google.com/spreadsheets/d/19KAdjIa6py2gVk5DrfFgqvmMr-jG1uvvVR0mz_LKF7c/edit?usp=sharing
